packages/website/src/components/editor/Editor.tsx

```diff
@@ -0,0 +1,329 @@
+import React from 'react';
+import type Monaco from 'monaco-editor';
+
+import { sandboxSingleton } from './loadSandbox';
+import { debounce } from '../lib/debounce';
+
+import type { Sandbox, SandboxConfig } from '../../vendor/sandbox';
+import type { WebLinter, TSESTree } from '@typescript-eslint/website-eslint';
+import type { ConfigModel, RuleDetails } from '../types';
+import { lintCode, LintCodeAction } from './lintCode';
+import { action } from './action';
+
+interface EditorProps extends ConfigModel {
```

Comment:
I think this's be a little easier to read with a `configModel: ConfigModel;` inside the props instead of extending. That way it can take in a prop more like `state={state}` the way `OptionsSelector` does.